File: Widhya/Project 2 - Uber Shortest Path Computation/C++ Version/Microtask2.cpp
#include <iostream>

using namespace std;

int distance_sum(int arr[], int n){

    // sort array using bubble sort (can be more efficient if used insertion sort)
    for(int i=0; i<n; i++){
        for(int j=i+1; j<n; j++){
            if(arr[j] < arr[i]){
                int temp= arr[j];
                arr[j] = arr[i];
                arr[i] = temp;
            }
        }
    }

    // Manhattan sum for each axis
    int res=0,sum=0;
    for(int i=0; i<n; i++){
        res+= arr[i]*i -sum;
        sum+= arr[i];
    }

    // return the result
    return res;
}

int totaldistancesum(int x[],int y[],int n){
    return distance_sum(x,n) + distance_sum(y,n);
}

int main()
{
    int n;
    cout << "Enter number of points > ";
    cin >> n;

    cout << endl;

    int x[n];
    for(int i=0; i<n; i++){
        cout << "Enter X" << i << " value > ";
        cin >> x[i];
    }

    cout << endl;

    int y[n];
    for(int i=0; i<n; i++){
        cout << "Enter Y" << i << " value > ";
        cin >> y[i];
    }

    cout << endl;

    cout << "Sum of Manhattan distance between all pairs of coordinates: " << totaldistancesum(x,y,n) << endl;


    return 0;
}

File: Widhya/Project 2 - Uber Shortest Path Computation/Python Version/microtask2.py
def distance_sum(arr, n):
    # sort the array
    arr.sort()

    # loop to find the distance of each independent axis
    res = 0
    sum = 0
    for i in range(n):
        res += arr[i] * i - sum
        sum += arr[i]

    return res


def totaldistancesum(x, y, n):
    return distance_sum(x, n) + distance_sum(y, n)


if __name__ == '__main__':
    n = int(input("Enter number of points > ")) # get array length
    print("") # for UI adjustment
    x = [float(input("Enter X point {} > ".format(_+1))) for _ in range(n)] # get x array
    print("") # for UI adjustment
    y = [float(input("Enter y Point {} > ".format(_ + 1))) for _ in range(n)] # get y array
    # print inputs & outputs
    print("\nEntered X coordinates list: {}".format(x))
    print("Entered y coordinates list: {}".format(y))
    print("Sum of Manhattan distance between all pairs of coordinates: {}".format(totaldistancesum(x,y,n)))